analysis/parameter_tuning.py

```diff
@@ -252,6 +261,118 @@ def tune(col,
+        1. Based on quantiles of privacy id contributions, candidates for
+        contribution bounding parameters chosen.
+        2. Utility analysis run for those parameters.
+        3. The best parameter set is chosen according to
+          options.minimizing_function.
+
+    The result contains output metrics for all utility analysis which were
+    performed.
+
+    Args:
+        col: collection where all elements are of the same type.
+          contribution_histograms:
+        backend: PipelineBackend with which the utility analysis will be run.
+        contribution_histograms: contribution histograms that should be
+         computed with compute_contribution_histograms().
+        options: options for tuning.
+        data_extractors: functions that extract needed pieces of information
+          from elements of 'col'. In case if the analysis performed on
+          pre-aggregated data, it should have type PreAggregateExtractors
+          otherwise DataExtractors.
+        public_partitions: A collection of partition keys that will be present
+          in the result. If not provided, tuning will be performed assuming
+          private partition selection is used.
+        return_per_partition: if true, it returns tuple, with the 2nd element
+          utility analysis per partitions.
+    Returns:
+        if return_per_partition == False:
+            returns 1 element collection which contains TuneResult
+        else returns tuple (1 element collection which contains TuneResult,
+        a collection which contains utility analysis results per partition).
+    """
+    _check_tune_args(options)
+
+    candidates = _find_candidate_parameters(contribution_histograms,
+                                            options.parameters_to_tune,
+                                            options.aggregate_params.metrics[0])
+
+    utility_analysis_options = analysis.UtilityAnalysisOptions(
+        epsilon=options.epsilon,
+        delta=options.delta,
+        aggregate_params=options.aggregate_params,
+        multi_param_configuration=candidates,
+        partitions_sampling_prob=options.partitions_sampling_prob,
+        pre_aggregated_data=options.pre_aggregated_data)
+    result = utility_analysis.perform_utility_analysis_new(
+        col, backend, utility_analysis_options, data_extractors,
+        public_partitions, return_utility_analysis_per_partition)
+    if return_utility_analysis_per_partition:
+        utility_result, per_partition_utility_result = result
+    else:
+        utility_result = result
+    # utility_result: (UtilityReport,)
+    # per_partition_utility_result: (pk, (PerPartitionMetrics))
+    use_public_partitions = public_partitions is not None
+
+    utility_result = backend.to_list(utility_result, "To list")
+    # 1 element collection with list[UtilityReport]
+    utility_result = backend.map(
+        utility_result,
+        lambda result: _convert_utility_analysis_to_tune_result_new(
```

Comment:
the difference with tune, only 
1.` backend.to_list`
2. and `_convert_utility_analysis_to_tune_result` -> `_convert_utility_analysis_to_tune_result_new`

Reply:
Why do we need the `backed.to_list`?